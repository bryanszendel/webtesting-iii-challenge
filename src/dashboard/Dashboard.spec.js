// Test away

import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react'

import Dashboard from './Dashboard';
import Display from '../display/Display'
import Controls from '../controls/Controls'

describe("<Dashboard />", () => {
  it("matches snapshot", () => {
    const tree = renderer.create(<Dashboard />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("cannot be closed or opened if it is locked", () => {
    const { getByText, findByText } = render(<Dashboard />)
    findByText(/locked/i)
    fireEvent.click(getByText("Close Gate"));
    fireEvent.click(getByText("Lock Gate"));
    fireEvent.click(getByText("Open Gate"));
    fireEvent.click(getByText("Open Gate"));
    findByText("Locked")
    findByText("Closed")
    findByText("Open Gate")
    findByText("Unlock Gate")
  })

  it("displays if gate is open/closed and if it is locked/unlocked", () => {
    let locked = false;
    let closed = false;

    const { getByText, findByText } = render(
      <Dashboard>
        <Display locked={locked} closed={closed} />
        <Controls />
      </Dashboard>);

    fireEvent.click(getByText("Close Gate"));
    findByText("Open Gate")
    fireEvent.click(getByText("Open Gate"));
    findByText("Close Gate")
  })

});