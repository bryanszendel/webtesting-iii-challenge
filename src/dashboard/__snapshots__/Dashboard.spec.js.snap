// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> matches snapshot 1`] = `
Array [
  <div
    className="display panel"
  >
    <div
      className="led green-led"
    >
      Unlocked
    </div>
    <div
      className="led green-led"
    >
      Open
    </div>
  </div>,
  <div
    className="controls panel"
  >
    <button
      className="toggle-btn"
      data-testid="closed"
      disabled={true}
      onClick={[Function]}
    >
      Lock Gate
    </button>
    <button
      className="toggle-btn"
      data-testid="locked"
      disabled={false}
      onClick={[Function]}
    >
      Close Gate
    </button>
  </div>,
]
`;
